machine/quanta/quanta_ly4r/kernel/fix-i2c-ismt-write-error.patch

```diff
@@ -0,0 +1,33 @@
+fix i2c-ismt write error
+
+diff --git a/drivers/i2c/busses/i2c-ismt.c b/drivers/i2c/busses/i2c-ismt.c
+index 6e932d1..7dd87b8 100644
+--- a/drivers/i2c/busses/i2c-ismt.c
++++ b/drivers/i2c/busses/i2c-ismt.c
+@@ -675,6 +675,23 @@ static irqreturn_t ismt_do_interrupt(int vec, void *data)
+  */
+ static irqreturn_t ismt_do_msi_interrupt(int vec, void *data)
+ {
++	u32 val;
++	struct ismt_priv *priv = data;
++
++	val = readl(priv->smba + ISMT_MSTR_MSTS);
++
++	while (val & (ISMT_MSTS_IP))
++	{
++		val = readl(priv->smba + ISMT_MSTR_MSTS);
++	}
```

Comment:
The usual idiom here is use a `do { .... } while();` construct.  Then you do not need the addition readl() done before the while().
```
do {
    val = readl(priv->smba + ISMT_MSTR_MSTS);
} while (val & (ISMT_MSTS_IP));
```